### ISSUE TITLE
boot-server.tsx no longer compiles with TypeScript 2.4 (React + Redux template)

### ISSUE THREAD

**Issue by chris-nissen** [aspnet/JavaScriptServices#1034]
### Title
boot-server.tsx no longer compiles with TypeScript 2.4 (React + Redux template)

### Functional impact
When webpacking in prod env mode, projects will fail TypeScript compilation. I'm not sure if any of the other templates are affected, but I expect they would be.

### Minimal repro steps
1. dotnet new reactredux
2. Update the version of typescript to "2.4.0" in package.json
3. npm install
4. node node_modules/webpack/bin/webpack.js --env.prod

### Expected result
Successful compilation.

### Actual result
TS2345: Argument of type '(params: BootFuncParams) => Promise<RenderResult>' is not assignable to parameter of type 'BootFunc'.
Type 'Promise<RenderResult>' is not assignable to type 'Promise<RenderToStringResult>'.
Type 'RenderResult' is not assignable to type 'RenderToStringResult'.
Type 'RedirectResult' is not assignable to type 'RenderToStringResult'.
Property 'html' is missing in type 'RedirectResult'.

### Further technical details
I think this can be fixed by updating the BootFunc interface like so:
```
interface BootFunc {
    (params: BootFuncParams): Promise<RenderToStringResult | RedirectResult>;
}
```

**Comment by KraigWalker** [comment]
If you typecast the function passed to `createServerRenderer` `as BootFunc` you should compile:

    export default createServerRenderer((params => {
    
        return new Promise<RenderResult>((resolve, reject) => {
    ...
    ...
        });
    }) as BootFunc);

**Comment by chris-nissen** [comment]
Yep, that works, thanks. I still hope someone will fix the d.ts, though.

**Comment by KraigWalker** [comment]
Might give this a looksy 😜 - I don't know how it is for everyone else, but TS 2.4 has made a lot of previously good code bad. Good thing it's meant to 😂

**Comment by KraigWalker** [comment]
Should probably wait though till 2.4 is out of rc

**Comment by DickvdBrink** [comment]
@DanielRosenwasser and @mhegazy, is this expected to break?

**Comment by SteveSandersonMS** [comment]
Thanks for reporting this! It's fixed in `aspnet-prerendering` 2.0.6 which is now published.

Also note that to make React typings work with TypeScript 2.4.0+, you must also update your copy of `@types/react` to 15.0.29 or later.
